src/binder/bind/bind_projection_clause.cpp

```diff
@@ -230,16 +229,28 @@ expression_vector Binder::bindOrderByExpressions(
     expression_vector exprs;
     for (auto& parsedExpr : parsedExprs) {
         auto expr = expressionBinder.bindExpression(*parsedExpr);
-        if (expr->dataType.getLogicalTypeID() == LogicalTypeID::NODE ||
-            expr->dataType.getLogicalTypeID() == LogicalTypeID::REL) {
-            throw BinderException(
-                "Cannot order by " + expr->toString() + ". Order by node or rel is not supported.");
+        if (!isOrderByKeyTypeSupported(expr->dataType)) {
+            throw BinderException(stringFormat("Cannot order by {}. Order by {} is not supported.",
+                expr->toString(), expr->dataType.toString()));
         }
         exprs.push_back(std::move(expr));
     }
     return exprs;
 }
 
+bool Binder::isOrderByKeyTypeSupported(const LogicalType& dataType) {
```

Comment:
Can u also open an issue about supporting more order by key types and assign to ziyi. I believe many of them can be naturally supported since we already support hash & compare

Reply:
I think it's more about how to encode different types of keys, but all should be doable.

Reply:
Updated the existing issue https://github.com/kuzudb/kuzu/issues/1212